sled-agent/src/config.rs

```diff
@@ -124,7 +124,7 @@ mod test {
     fn test_smf_configs() {
         let manifest = std::env::var("CARGO_MANIFEST_DIR")
             .expect("Cannot access manifest directory");
-        let smf = PathBuf::from(manifest).join("../smf/sled-agent");
+        let smf = Utf8PathBuf::from(manifest).join("../smf/sled-agent");
 
         let mut configs_seen = 0;
         for variant in std::fs::read_dir(smf).unwrap() {
```

Comment:
One option here is to use [`read_dir_utf8`](https://camino-rs.github.io/camino/rustdoc/camino/struct.Utf8Path.html#method.read_dir_utf8). However, that will fail if any of the paths are non-UTF-8.

* You can keep iterating over the even if one of the entries fails to parse as valid UTF-8.
* This code uses unwraps on I/O errors already, so maybe these additional kinds of errors are not a huge deal.

Reply:
Done in https://github.com/oxidecomputer/omicron/pull/2988/commits/ced5dc92aa41c74da6a846aaa9cd59d20d9c7843

Given that these are tests, I'm totally fine bailing if we hit something non-UTF-8